apps/studio/components/interfaces/ProjectCreation/OrganizationSelector.tsx

```diff
@@ -0,0 +1,89 @@
+import { PermissionAction } from '@supabase/shared-types/out/constants'
+import { useRouter } from 'next/router'
+import { UseFormReturn } from 'react-hook-form'
+
+import { useParams } from 'common'
+import Panel from 'components/ui/Panel'
+import { useOrganizationsQuery } from 'data/organizations/organizations-query'
+import { useAsyncCheckPermissions } from 'hooks/misc/useCheckPermissions'
+import { useSelectedOrganizationQuery } from 'hooks/misc/useSelectedOrganization'
+import {
+  Badge,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Select_Shadcn_,
+  SelectContent_Shadcn_,
+  SelectGroup_Shadcn_,
+  SelectItem_Shadcn_,
+  SelectTrigger_Shadcn_,
+  SelectValue_Shadcn_,
+} from 'ui'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+import { OrgNotFound } from '../Organization/OrgNotFound'
+import { NotOrganizationOwnerWarning } from './NotOrganizationOwnerWarning'
+import { CreateProjectForm } from './ProjectCreation.schema'
+
+interface OrganizationSelectorProps {
+  form: UseFormReturn<CreateProjectForm>
+}
+
+export const OrganizationSelector = ({ form }: OrganizationSelectorProps) => {
+  const router = useRouter()
+  const { slug } = useParams()
+  const { data: currentOrg } = useSelectedOrganizationQuery()
+  const { can: isAdmin } = useAsyncCheckPermissions(PermissionAction.CREATE, 'projects')
+
+  const { data: organizations, isSuccess: isOrganizationsSuccess } = useOrganizationsQuery()
+  const isInvalidSlug = isOrganizationsSuccess && currentOrg === undefined
+  const orgNotFound = isOrganizationsSuccess && (organizations?.length ?? 0) > 0 && isInvalidSlug
+
+  return (
+    <Panel.Content className="space-y-4">
+      {isAdmin && !isInvalidSlug && (
+        <FormField_Shadcn_
+          control={form.control}
+          name="organization"
+          render={({ field }) => (
+            <FormItemLayout label="Organization" layout="horizontal">
+              {(organizations?.length ?? 0) > 0 && (
```

Comment:
Are we missing a loading case and an empty case here?

Reply:
it's technically handled in the /new/[slug] page within panel where the loading state of the UI is dependent on `isSuccessOrganizations`
<img width="597" height="184" alt="image" src="https://github.com/user-attachments/assets/55ecc76c-3ee1-4ec8-b7ac-62471fd905be" />